libraries/core_libs/network/src/tarcap/shared_states/pbft_syncing_state.cpp

```diff
@@ -65,4 +65,32 @@ bool PbftSyncingState::isPbftSyncing() {
   return pbft_syncing_;
 }
 
+bool PbftSyncingState::updatePeerAskingPeriod(const dev::p2p::NodeID& peer_id, uint64_t period) {
+  auto now = std::chrono::steady_clock::now();
+
+  UpgradableLock lock(asking_period_mutex_);
+  if (asking_period_peers_table_.contains(peer_id) && asking_period_peers_table_[peer_id].first >= period &&
+      std::chrono::duration_cast<std::chrono::milliseconds>(now - asking_period_peers_table_[peer_id].second) <=
+          kPeerAskingPeriodTimeout) {
+    return false;
+  }
+
+  UpgradeLock locked(lock);
+  asking_period_peers_table_[peer_id] = std::make_pair(period, now);
+
+  return true;
+}
+
+bool PbftSyncingState::updatePeerSyncingPeriod(const dev::p2p::NodeID& peer_id, uint64_t period) {
+  UpgradableLock lock(incoming_blocks_mutex_);
```

Comment:
UpgradableLock + UpgradeLock design make sense only when there is another "reader" (shared_lock) otherwise there is no gain, as UpgradableLock can not be locked 2x. So I would get rid of it and keep it simple with unique_lock

Reply:
I would say we should not use this pattern at all...

Reply:
not sure about that, there are several places in our code where "heavy" computing is done in `UpgradableLock` and there is no need to completely lock it as it blocks all the "reader" meanwhile

Reply:
Change to std::unique_lock